sqlmesh/core/snapshot/evaluator.py

```diff
@@ -407,6 +407,19 @@ def _create_snapshot(
         evaluation_strategy = _evaluation_strategy(snapshot, self.adapter)
 
         with self.adapter.transaction(TransactionType.DDL), self.adapter.session():
+            if is_dev and not snapshot.previous_versions:
```

Comment:
why do you need this section here, shouldn't all snapshots create a table?

Reply:
is it because this snapshot is forward only and we don't create tables on forward only snapshots?